web/web.go

```diff
@@ -202,38 +236,36 @@ func New(logger log.Logger, o *Options) *Handler {
 		router = router.WithPrefix(o.RoutePrefix)
 	}
 
-	instrh := prometheus.InstrumentHandler
-	instrf := prometheus.InstrumentHandlerFunc
 	readyf := h.testReady
 
 	router.Get("/", func(w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, path.Join(o.ExternalURL.Path, "/graph"), http.StatusFound)
 	})
 
-	router.Get("/alerts", readyf(instrf("alerts", h.alerts)))
-	router.Get("/graph", readyf(instrf("graph", h.graph)))
-	router.Get("/status", readyf(instrf("status", h.status)))
-	router.Get("/flags", readyf(instrf("flags", h.flags)))
-	router.Get("/config", readyf(instrf("config", h.serveConfig)))
-	router.Get("/rules", readyf(instrf("rules", h.rules)))
-	router.Get("/targets", readyf(instrf("targets", h.targets)))
-	router.Get("/version", readyf(instrf("version", h.version)))
-	router.Get("/service-discovery", readyf(instrf("servicediscovery", h.serviceDiscovery)))
+	router.Get("/alerts", readyf(instrumentHandler("alerts", h.alerts)))
```

Comment:
Why do you have the instrumentHandler here rather than using WithInstrumentation?

Reply:
Because of the other handlers like `/-/*` and `/debug/*` which aren't instrumented ([link](https://github.com/simonpasquier/prometheus/blob/2db058636409e32d7ab2cbb4dd00ea7095d9aa94/web/web.go#L271-L303)).

Reply:
I don't see any harm in instrumenting them.